File: dart/test/stack/stack_test.dart
import 'package:test/test.dart';

import '../../src/stack/Stack.dart';

void main() {
  test("stack is empty", () {
    var stack = Stack();
    expect(stack.isEmpty(), true);
  });

  test("stack is not empty", () {
    var stack = Stack();
    stack.push(1);
    expect(stack.isEmpty(), false);
    expect(stack.top(), 1);
  });

  test("stack push", () {
    var stack = Stack();
    stack.push(1);
    stack.push(2);
    stack.push(3);
    expect(stack.top(), 3);
    expect(stack.size(), 3);
  });

  test("stack pop", () {
    var stack = Stack();
    stack.push(1);
    stack.push(2);
    stack.push(3);
    expect(stack.pop(), 3);
    expect(stack.pop(), 2);
    expect(stack.pop(), 1);
    expect(stack.isEmpty(), true);
  });

  test("stack top", () {
    var stack = Stack();
    stack.push(1);
    stack.push(2);
    stack.push(3);
    expect(stack.top(), 3);
    expect(stack.size(), 3);
    expect(stack.isEmpty(), false);
    expect(stack.pop(), 3);
    expect(stack.top(), 2);
    expect(stack.pop(), 2);
    expect(stack.top(), 1);
    expect(stack.pop(), 1);
    expect(stack.isEmpty(), true);
  });
}

File: Java/src/test/java/stack/StackTest.java
package stack;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import com.example.stack.Stack;

public class StackTest {
    
    @Test
    public void testIsEmpty() {
        Stack stack = new Stack();
        Assertions.assertEquals(true, stack.isEmpty());

    }

    @Test
    public void testPush() {
        Stack stack = new Stack();
        stack.push(10);
        Assertions.assertEquals(10, stack.top());
        Assertions.assertEquals(1, stack.size());
        Assertions.assertEquals(false, stack.isEmpty());

        stack.push(20);
        stack.push(30);
        Assertions.assertEquals(30, stack.top());
        Assertions.assertEquals(3, stack.size());

    }

    @Test
    public void testPop() {
        Stack stack = new Stack();
        stack.push(10);
        Assertions.assertEquals(10, stack.pop());
        Assertions.assertEquals(0, stack.size());
        Assertions.assertEquals(true, stack.isEmpty());

        stack.push(10);
        stack.push(20);
        Assertions.assertEquals(20, stack.pop());
        Assertions.assertEquals(1, stack.size());
        Assertions.assertEquals(false, stack.isEmpty());
        Assertions.assertEquals(10, stack.pop());
    }
}
